### ISSUE TITLE
(Question) : How to compare all the pictures from one folder?

### ISSUE THREAD

**Issue by Jean-Jerome-Casalonga-i2** [serengil/deepface#572]
Hi, I hope you won't mind me asking a question here, as it's not an issue...

I've a folder with several thousands pictures, and I'd like to find when they represent the same persons.
I understand that the basic principal is to compare each pair of pictures:
`result = DeepFace.verify(img1_path = "img1.jpg", img2_path = "img2.jpg")
`
But this means that for each test, DeepFace must do Detect, Align, Normalize and Represent., to finaly be able to Verify.
But if it does this for each comparison between 2 pictures, each picture will be compared multiple times.

So, I understand (and I hope I'm correct) that I must use Embeddings.
I'm doing a first pass, embedding all the pictures in the folder.
But then, what's the next step ?
How do I use the Embedding vectors of all the pictures to compare them eachother ?

Basically, there should be a nested loop:

 For i = 0 to number of picture, do
      For j = i + 1 to Number of pictures, do
          Compare vector(i) with Vector(j)
     end Loop j
 End Loop i

What is the correct way to do this ?

 Thanks a lot

**Comment by serengil** [comment]
that is the correct way as you mentioned

**Comment by serengil** [comment]
if the distance is less than a threshold value, then you will classify it as same person. you can find the tuned thresholds here: https://github.com/serengil/deepface/blob/56adc351f22047c5c70fcf92cd97130ba27a707c/deepface/commons/distance.py#L28
